src/signing.rs

```diff
@@ -208,8 +209,11 @@ impl SigningKey {
     ///
     /// # Examples
     ///
-    #[cfg_attr(feature = "rand_core", doc = "```")]
-    #[cfg_attr(not(feature = "rand_core"), doc = "```ignore")]
+    #[cfg_attr(all(feature = "rand_core", feature = "digest"), doc = "```")]
+    #[cfg_attr(
+        any(not(feature = "rand_core"), not(feature = "digest")),
+        doc = "```ignore"
+    )]
     /// use ed25519_dalek::Digest;
     /// use ed25519_dalek::SigningKey;
     /// use ed25519_dalek::Sha512;
```

Comment:
So are we supposed to re-export Sha512 ? If not may neet to remove this here.

Reply:
How were these passing before?

I think it's ok to gate a re-export behind the `digest` feature.

Reply:
Why were we exporting `Sha512` at all? I'm not entirely opposed, it just seems like an odd choice. People should probably bring their own digest function. The fact we use Sha512 under the hood is more an implementation detail.

Reply:
It's indeed a mystery how these were passing before as Sha512 was never exported.
It's also complicated since the Digest'd output type comes essentially from Sha512 which we require as input and there is whole lot of back/forward related coupling related to that.
I would say that makes it feasible to export for compatibility reasons as your dependency tree might have incompat types beween versions when Digest is used and using the exported Sha512 gives the right Digest input.
That said this wasn't exported before but I think if we are using it in the API gating it's may usually be best to export.

Reply:
The plot thickens.. the Sha512 was there always.. guess why? src/lib.rs

```rust
pub use crate::constants::*;
pub use crate::errors::*;
pub use crate::signing::*;
pub use crate::verifying::*;

```... as this also exported Sha512 guess from where ? :D

Reply:
spoiler: it came from verifying.rs which did pub re-export of Sha512 that then got re-exported at lib.rs